src/DynamoCore/Graph/Workspaces/PackageDependencyInfo.cs

```diff
@@ -175,7 +175,18 @@ internal class PackageDependencyInfo : INodeLibraryDependencyInfo, IPackageInfo
         /// <summary>
         /// Version of the package
         /// </summary>
-        public Version Version => PackageInfo.Version;
+        public Version Version
+        {
+            get
+            {
+                return PackageInfo.Version;
+            }
+            set
+            {
+                if(PackageInfo.Version != value)
+                    PackageInfo.Version = value;
```

Comment:
This definitely works but unfortunately is the wrong way for sure. I would like to discuss with the team the right approach here, one way I can think of is have an internal method to update it and make the assembly internal visible to dependency view extension only. Unless we make it visible to any extension. What do you think @DynamoDS/dynamo

Reply:
I'm sorry @QilongTang I'm confused, why do you need to do this - why does this need a setter at all? Can't we just make a new PackageDependencyInfo object?

Reply:
Good point, let me try

Reply:
@mjkkirschner That way it is even more complex than this approach, because we will need to map the nodes property which is also read-only, and will need to switch the old info out and replace it with the new one. The whole process is not possible until we expose the setter for nodes which is not a good trade off.

Reply:
I think you should be careful before doing this because it breaks some invariants made previously -
The same way that now updating this is useful because the nodeInfo objects have been updated - can also be harmful - because some code may have relied on this PackageInfo not changing - 

I'm worried specifically about the customNode package versioning check stuff.

Reply:
@mjkkirschner I guess we need to also update the version CBN in this case.. seems whether or not create a new info object does not matter but we need to make sure all the record in memory should be aligned...